### PR TITLE
Define Time.__bool__ and add tests (closes #3520)

### LINKED ISSUE #3520: time.Time __bool__ and __len__ unexpected behaviour
I noticed this behaviour of time.Time in 0.4.4 and was able to replicate it in 1.0:
````python
>>> t = time.Time('2015-01-01')
>>> len(t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/michele/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/astropy/time/core.py", line 940, in __len__
    return len(self.jd1)
TypeError: object of type 'float' has no len()
````
For behaviour consistent with the standard Python `datetime.datetime`, `len(t)` should instead throw an `AttributeError`.

This further comes in when one tries:
````python
>>> if t:
...     print(t)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/Users/michele/Library/Enthought/Canopy_64bit/User/lib/python2.7/site-packages/astropy/time/core.py", line 940, in __len__
    return len(self.jd1)
TypeError: object of type 'float' has no len()
````

For behaviour consistent with `datetime`, `bool(t)` for a `time.Time` would always return `True`.

### PR DESCRIPTION
**Pull request by mhvk**
This ensures any time object evaluates to `True` unless it is empty (see #3520). Also adds tests for both `__bool__` and `__len__`.

### CODE REVIEW

**Comment by mhvk** [comment]
Added a more informative error for getting the length of a scalar `Time` object. (also cancelled travis build; do not know how to do that for appveyor)

**Comment by taldcroft** [comment]
Apart from my minor comment (which you can fix or not), this looks good.  Just needs a CHANGES.rst update.

**Comment by mhvk** [comment]
@taldcroft - I don't see a comment!?

File: astropy/time/tests/test_basic.py [hunk 1]
```diff
@@ -746,6 +750,31 @@ def test_dir():
...
+    assert len(t1) == 2 and t1.size == 10
+    # Can have length 1 or length 0 arrays.
+    t2 = t[:1]
+    assert len(t2) == 1 and t2.size == 1
+    t3 = t[:0]
+    assert len(t3) == 0 and t3.size == 0
+    # But cannot get length from scalar.
+    t4 = t[0]
+    with pytest.raises(TypeError):
+        len(t4)
```
Comment (taldcroft) [inline astropy/time/tests/test_basic.py]: Not absolutely required but it's always nice to check the error message to make sure you are getting the expected `"Scalar {0} object has no len()"`.

**Comment by mhvk** [comment]
@taldcroft - OK, I added a very simple test for the error message and updated `CHANGES.rst`. Think this is ready.

**Comment by astrofrog** [comment]
@mhvk - the changelog entry should be added to a new 1.0.1 section

**Comment by astrofrog** [comment]
@mhvk - I've now added empty sections to the changelog in master so if you rebase this PR then you can add the changelog entry to the right section

**Comment by mhvk** [comment]
@astrofrog - ok, moved to the right section (skipping travis).

**Comment by mhvk** [comment]
@taldcroft, @astrofrog - shall we merge this?

**Comment by taldcroft** [comment]
Looks good to me.

**Pull request merged**